### PR TITLE
Feat: Update user personal details

### LINKED ISSUE #47: Update User Personal Details
## Description
As a user,
I need to update the personal details ,
so that I can change my personal details information.

## Mocks
NA

### Update [Required]
- [ ] Create an "Update" functionality for user personal details page

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
1 day

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description
Allow user to update their personal details information

Fixes #47 

### Type of Change:
- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- login as a user
- go to `My Space` > `Profile` > `Personal details`
Screen Shot 2020-07-27 at 2 04 38 am

Screen Shot 2020-07-27 at 2 04 45 am

- Update details including useername (**NOTE** this is a bug on MS. It's already reported on [issue#596](https://github.com/anitab-org/mentorship-backend/issues/596). User should be allowed to update details without changing username)
- see success message on the bottom of the form
- refresh the page or logout and login with same username. This should fail
Screen Shot 2020-07-27 at 9 42 51 am

- login with new username and go to Personal Details page. See new details
Screen Shot 2020-07-27 at 9 43 19 am

Screen Shot 2020-07-27 at 9 47 28 am

### Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . Here's the PR with update user personal details functionality. Please review when you have time. Ta

File: src/myspace/PersonalDetails.jsx [hunk 1]
```diff
@@ -6,8 +6,12 @@ import "./MySpace.css";
 
 export default function PersonalDetails() {
   const [errorMessage, setErrorMessage] = useState(null);
+  const [responseMessage, setResponseMessage] = useState(null);
   const [personalDetails, setPersonalDetails] = useState({});
   const { access_token } = useContext(AuthContext);
+  const [validName, setValidName] = useState(true);
```
Comment (meenakshi-dhanani) [inline src/myspace/PersonalDetails.jsx]: use isValidName to name booleans. 
booleans are named as isSomething, hasSomething

File: src/myspace/PersonalDetails.jsx [hunk 2]
```diff
@@ -112,23 +187,23 @@ export default function PersonalDetails() {
                           <div className="row">
                             <div className="col-sm">
                               <input
-                                name="mentor"
+                                name="available_to_mentor"
                                 aria-label="mentor"
                                 type={type}
                                 defaultChecked={personalDetails.available_to_mentor}
+                                value={"on" ? true : false}
                                 id={`inline-${type}-1`}
```
Comment (meenakshi-dhanani) [inline src/myspace/PersonalDetails.jsx]: why have the id?

File: src/myspace/PersonalDetails.jsx [hunk 3]
```diff
@@ -18,23 +22,70 @@ export default function PersonalDetails() {
...
 
   });
 
+  const handleSubmit = async e => {
+    e.preventDefault();
+
+    let payload = {}
+    new FormData(e.target).forEach((value, key) => {
+      if (key === "need_mentoring" || key === "available_to_mentor")
+          if (value === "true")
```
Comment (meenakshi-dhanani) [inline src/myspace/PersonalDetails.jsx:46]: Too many ifs and else's here. Code doesn't look clean with so many conditionals

Comment (meenakshi-dhanani) [reply src/myspace/PersonalDetails.jsx:46]: We can refactor this tomorrow together, or give it a shot if you like

File: src/myspace/PersonalDetails.jsx (hunk 1 above)
Comment (mtreacy002) [reply src/myspace/PersonalDetails.jsx]: Done. Refactored to IsValidName and isValidUsername

File: src/myspace/PersonalDetails.jsx (hunk 2 above)
Comment (mtreacy002) [reply src/myspace/PersonalDetails.jsx]: Done. removed the ids on checkboxes

File: src/myspace/PersonalDetails.jsx (hunk 3 above)
Comment (mtreacy002) [reply src/myspace/PersonalDetails.jsx:46]: For this one, I'll wait till our meeting tonight on your office hour (if you feel well enough). I'm not sure what else to do with this option.

**Review by meenakshi-dhanani — APPROVED** [review]

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _Update user personal details (username already exists)_  
      _Screenshot/gif/url_:  

       test-pr-54-update-personal-details-username-details

     _Expected Result_:  As a user, I should be able to update my personal details (if the username is not changed, then it will show `A user with that username already exists.` (error from MS)
     _Actual Result_: Same as expected


   * _Update user personal details by changing username_  
      _Screenshot/gif/url_:  

       
       test-pr-54-update-personal-details


     _Expected Result_:  As a user, I should be able to update my personal details and no error shows up when I change my username.
     _Actual Result_:  Same as expected.

3. OS Version: Windows 10

**Pull request merged**